tests/lean/run/localization.lean

```diff
@@ -0,0 +1,1036 @@
+  { (Function.Surjective.addSemigroup _ Quotient.surjective_Quotient_mk'' sorry :
+      AddSemigroup c.Quotient) with
+    toAdd := AddCon.hasAdd _ }
+
+instance addMonoid {M : Type _} [AddMonoid M] (c : AddCon M) : AddMonoid c.Quotient :=
+  { (Function.Surjective.addMonoid _ Quotient.surjective_Quotient_mk'' sorry sorry sorry : AddMonoid c.Quotient) with
+    toAddSemigroup := AddCon.addSemigroup _
+    toZero := AddCon.zero _ }
+
+end Monoids
+
+variable [AddGroup M] [AddGroup N] [AddGroup P] (c : AddCon M)
+
+instance hasNeg : Neg c.Quotient :=
+  ⟨(Quotient.map' Neg.neg) sorry⟩
+
+instance hasSub : Sub c.Quotient :=
+  ⟨(Quotient.map₂' (· - ·)) sorry⟩
+
+instance _root_.AddCon.Quotient.zsmul {M : Type _} [AddGroup M] (c : AddCon M) :
+    SMul Int c.Quotient :=
+  ⟨fun z => (Quotient.map' (z • ·)) sorry⟩
+
+instance addGroup : AddGroup c.Quotient :=
+  { (Function.Surjective.addGroup Quotient.mk''
+      Quotient.surjective_Quotient_mk'' sorry sorry sorry : AddGroup c.Quotient) with
+    toAddMonoid := AddCon.addMonoid _
+    toNeg := AddCon.hasNeg _
+    toSub := AddCon.hasSub _ }
+
+end AddCon
+
+end Mathlib.GroupTheory.Congruence
+
+section Mathlib.GroupTheory.GroupAction.Opposite
+
+variable {α}
+
+instance Add.toHasOppositeVAdd [Add α] : VAdd αᵃᵒᵖ α :=
+  ⟨fun c x => x + c.unop⟩
+
+instance AddMonoid.toOppositeVAdd [AddMonoid α] : VAdd αᵃᵒᵖ α where
+  vadd := (· +ᵥ ·)
+
+end Mathlib.GroupTheory.GroupAction.Opposite
+
+section Mathlib.GroupTheory.Subgroup.Basic
+
+variable {A : Type _} [AddGroup A]
+
+structure AddSubgroup (G : Type _) [AddGroup G] extends AddSubmonoid G where
+
+namespace AddSubgroup
+
+instance : SetLike (AddSubgroup A) A where
+  coe s := s.carrier
+
+end AddSubgroup
+
+end Mathlib.GroupTheory.Subgroup.Basic
+
+section Mathlib.GroupTheory.GroupAction.Basic
+
+universe u v
+
+namespace VAdd
+
+variable (M : Type u) [AddMonoid M] {α : Type v} [VAdd M α]
+
+def orbit (a : α) :=
+  Set.range fun m : M => m +ᵥ a
+
+variable (G α : Type _) [AddGroup G] [VAdd G α]
+
+def orbitRel : Setoid α where
+  r a b := a ∈ orbit G b
+  iseqv := sorry
+
+end VAdd
+
+end Mathlib.GroupTheory.GroupAction.Basic
+
+section Mathlib.GroupTheory.Subgroup.MulOpposite
+
+variable {G : Type _} [AddGroup G]
+
+protected def AddSubgroup.op (H : AddSubgroup G) : AddSubgroup Gᵃᵒᵖ where
+  carrier := MulOpposite.unop ⁻¹' (H : Set G)
+
+end Mathlib.GroupTheory.Subgroup.MulOpposite
+
+section Mathlib.Algebra.Quotient
+
+universe u v
+
+class HasQuotient (A : outParam <| Type u) (B : Type v) where
+  quotient' : B → Type max u v
+
+def HasQuotient.Quotient (A : outParam <| Type u) {B : Type v}
+    [HasQuotient A B] (b : B) : Type max u v :=
+  HasQuotient.quotient' b
+
+notation:35 G " ⧸ " H:34 => HasQuotient.Quotient G H
+
+end Mathlib.Algebra.Quotient
+
+section Mathlib.GroupTheory.Coset
+
+variable {α : Type _}
+
+namespace QuotientAddGroup
+
+variable [AddGroup α] (s : AddSubgroup α)
+
```

Comment:
```suggestion
instance [Add α] : VAdd α α := 
  { vadd := Add.add }

def leftRel : Setoid α :=
  VAdd.orbitRel s α
```
- makes this very fast 
- explains why making `op` irreducible speeds things up 
- fixes defeq abuse 

The instance is just so you can switch out `s.op` for `s`.